core/data/src/main/java/uk/gov/gchq/gaffer/data/element/function/PropertiesFilter.java

```diff
@@ -0,0 +1,167 @@
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.data.element.function;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import uk.gov.gchq.gaffer.commonutil.ToStringBuilder;
+import uk.gov.gchq.gaffer.data.element.Properties;
+import uk.gov.gchq.koryphe.Since;
+import uk.gov.gchq.koryphe.Summary;
+import uk.gov.gchq.koryphe.ValidationResult;
+import uk.gov.gchq.koryphe.tuple.predicate.TupleAdaptedPredicate;
+import uk.gov.gchq.koryphe.tuple.predicate.TupleAdaptedPredicateComposite;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * An {@code PropertiesFilter} is a {@link Predicate} which evaluates a condition against
+ * a provided {@link Properties} object.
+ */
+@JsonPropertyOrder(alphabetic = true)
+@Since("1.21.0")
+@Summary("A Predicate which evaluates against a Properties")
+public class PropertiesFilter extends TupleAdaptedPredicateComposite<String> {
+    private final PropertiesTuple propertiesTuple = new PropertiesTuple();
+    private boolean readOnly;
+
+    public boolean test(final Properties properties) {
+        propertiesTuple.setProperties(properties);
+        return test(propertiesTuple);
+    }
+
+    public ValidationResult testWithValidationResult(final Properties properties) {
+        final ValidationResult result = new ValidationResult();
+        propertiesTuple.setProperties(properties);
+        components.stream()
+                .filter(predicate -> !predicate.test(propertiesTuple))
+                .forEach(predicate -> result.addError(getErrorMsg(predicate)));
+        return result;
+    }
+
+    private String getErrorMsg(final TupleAdaptedPredicate<String, ?> predicate) {
+        final StringBuilder builder = new StringBuilder();
+        builder.append("Filter: ")
+                .append(predicate.getPredicate())
+                .append(" returned false for properties: {");
+
+        boolean firstProp = true;
+        for (final String selection : predicate.getSelection()) {
+            final Object value = propertiesTuple.get(selection);
+            final String valueStr = null != value ? String.format("<%s>%s", value.getClass().getCanonicalName(), value) : "null";
+            if (firstProp) {
+                firstProp = false;
+            } else {
+                builder.append(", ");
+            }
+            builder.append(selection)
+                    .append(": ")
+                    .append(valueStr);
+        }
+        builder.append("}");
+
+        return builder.toString();
+    }
+
+    @Override
+    public List<TupleAdaptedPredicate<String, ?>> getComponents() {
+        if (readOnly) {
+            return Collections.unmodifiableList(super.getComponents());
+        }
+
+        return super.getComponents();
+    }
+
+    public void lock() {
+        readOnly = true;
+    }
```

Comment:
Very generally comment, but I would expect to see a matching unlock. It seems strange that once locked there is no way to reverse this.

Reply:
Hmm you're right, seems it is the same as how `ElementFilter` currently works: 

https://github.com/gchq/Gaffer/blob/b600adedaebc9d7eb8d2dd44da1b05b98fd7bc7f/core/data/src/main/java/uk/gov/gchq/gaffer/data/element/function/ElementFilter.java#L90-L93

I wonder if there is a reason for this